metricbeat/module/haproxy/info/data.go

```diff
@@ -145,14 +145,16 @@ func eventMapping(info *haproxy.Info) (common.MapStr, error) {
 			// Convert this value to a float between 0.0 and 1.0
 			fval, err := strconv.ParseFloat(f.Interface().(string), 64)
 			if err != nil {
-				return nil, err
+				r.Error(err)
```

Comment:
In the rest of the metricset you are wrapping errors and giving them more context. Not here?

Reply:
This PR is keeping all the behaviour as is as it's required for ECS. Agree that we need to do some more cleanup and do better with error message. Perhaps we can add a general item about error handling here?